pkg/endpoint/endpoint.go

```diff
@@ -441,6 +447,11 @@ func createEndpoint(owner regeneration.Owner, proxy EndpointProxy, allocator cac
 		controllers:     controller.NewManager(),
 		regenFailedChan: make(chan struct{}, 1),
 		allocator:       allocator,
+		mcastManager:    mcastmanager.New(""),
+	}
+
+	if option.Config.EnableIPv6NDP && len(option.Config.IPv6MCastDevice) > 0 {
```

Comment:
Since you check earlier that `EnableIPv6NDP` always comes with `IPv6MCastDevice`, we can get rid of the second part of the condition here I think.